$NetBSD: patch-aa,v 1.2 2010/07/18 00:58:51 schnoebe Exp $

#
# Change to use the CMake function to find the python interpreter,
# rather than assuming it is "python".  Trying to push upstream.
#
# change man path to be configurable to PKGSRC standards.
#

--- CMakeLists.txt.orig	2010-07-13 17:38:39.000000000 +0000
+++ CMakeLists.txt
@@ -21,8 +21,11 @@ set(libpoco_DIR "${CMAKE_SOURCE_DIR}/cma
 find_package(libpoco REQUIRED)
 set(cppunit_DIR "${CMAKE_SOURCE_DIR}/cmake_modules")
 find_package(cppunit)
+
+# correctly find the python interpreter
+include(FindPythonInterp)
  
-execute_process ( COMMAND python -c "from distutils.sysconfig import get_python_lib; print get_python_lib()" OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process ( COMMAND ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print get_python_lib()" OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
 message(STATUS "Python site-packages directory: " ${PYTHON_SITE_PACKAGES})
 
 include_directories(${GLOOX_INCLUDE_DIR})
@@ -362,7 +365,7 @@ ELSE(NOT GETTEXT_MSGFMT_EXECUTABLE)
  
 ENDIF(NOT GETTEXT_MSGFMT_EXECUTABLE)
 
-ADD_CUSTOM_TARGET (man ALL python ${CMAKE_CURRENT_SOURCE_DIR}/spectrumctl/make_man_page.py > ${CMAKE_CURRENT_SOURCE_DIR}/man/spectrumctl.8)
+ADD_CUSTOM_TARGET (man ALL ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/spectrumctl/make_man_page.py > ${CMAKE_CURRENT_SOURCE_DIR}/man/spectrumctl.8)
 
 
 INSTALL(TARGETS spectrum
@@ -395,11 +398,11 @@ INSTALL(FILES
 
 INSTALL(FILES
     ${CMAKE_CURRENT_SOURCE_DIR}/man/spectrumctl.8
-	DESTINATION share/man/man8
+	DESTINATION @PKGMANDIR@/man8
     )
 INSTALL(FILES
     ${CMAKE_CURRENT_SOURCE_DIR}/man/spectrum.cfg.5
-	DESTINATION share/man/man5
+	DESTINATION @PKGMANDIR@/man5
     )
 
 endif (NOT WITHOUT_SPECTRUMCTL)
